org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/refactoring2/ReplaceFileContentChange.java

```diff
@@ -0,0 +1,109 @@
+/**
+ * Copyright (c) 2017, 2020 TypeFox GmbH (http://www.typefox.io) and others.
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.xtext.ui.refactoring2;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.InputStream;
+
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.resources.IResource;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.OperationCanceledException;
+import org.eclipse.ltk.core.refactoring.Change;
+import org.eclipse.ltk.core.refactoring.RefactoringStatus;
+import org.eclipse.ltk.core.refactoring.resource.ResourceChange;
+import org.eclipse.xtext.xbase.lib.Exceptions;
+import org.eclipse.xtext.xbase.lib.Pure;
+
+/**
+ * @author koehnlein - Initial contribution and API
+ * @since 2.13
+ */
+public class ReplaceFileContentChange extends ResourceChange {
+
+	private final IFile file;
+
+	private final byte[] newContents;
+
+	private String name;
+
+	public ReplaceFileContentChange(String name, IFile file, byte[] newContents) {
+		this(file, newContents);
+		this.name = name;
+	}
+
+	public ReplaceFileContentChange(final IFile file, final byte[] newContents) {
+		super();
+		this.file = file;
+		this.newContents = newContents;
+	}
+
+	@Override
+	protected IResource getModifiedResource() {
+		return file;
+	}
+
+	@Override
+	public RefactoringStatus isValid(IProgressMonitor pm) throws CoreException, OperationCanceledException {
+		final RefactoringStatus result = new RefactoringStatus();
+		if (!file.exists()) {
+			result.addFatalError("File " + file.getFullPath() + " does not exist");
+		}
+		return result;
+	}
+
+	@Override
+	public Change perform(final IProgressMonitor pm) throws CoreException {
+		final byte[] oldContents = this.getOldContents();
+		final ByteArrayInputStream newContentsIS = new ByteArrayInputStream(this.newContents);
+		return TryWithResource.<ReplaceFileContentChange> tryWith(newContentsIS, () -> {
```

Comment:
No need for the generic

Reply:
Also no need for the `TryWithResource` utility.

Reply:
@szarnekow If I don't use the utility here, I'll be forced to duplicate the following piece of code (abstracted in the utility). 
```
try {                                      
	newContentsIS.close();                     
} catch (Throwable t) {                    
	LOG.error("Error closing stream", t);  
}                                          
```

Reply:
The throwable should be propagated as a CoreException rather than being only logged.

Reply:
@szarnekow Do you want me to import and use the utility statically, or you want me not to use the utility here?

Reply:
I want to get rid of `org.eclipse.xtext.ui.refactoring2.TryWithResource`